### ISSUE TITLE
Check on start of execution of Elgg upgrade if all necessary prerequisites are fullfilled

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#10140]
Right now it might happen (by mistake or lack of knowledge) that you end with a non-working site after running a site upgrade or even worse the site upgrade gets stuck at some point in between.

For example, the upgrade could fail to complete if a site admin tries to upgrade directly to the latest available Elgg version omitting one or more minor releases in between. This might also happen when using Softaculus (as maybe specifically used by people new to Elgg without the necessary knowledge about the upgrade process).

Or you might not have the necessary php version the new Elgg release requires installed on the server yet.

I think it would be good if the install process would start with checking if these kind of prerequisites are matched before starting to make any changes in the database or data directory. Other prerequisites that I've not mentioned above might also be useful to check. But I think it should really be done before the real start of the upgrade process because it's still possible to restore the old state of the Elgg install directory at this point (at least with a backup available) whereas it's not as easy anymore if the database and/or data directory have already been modified or if the server does not fulfill the requirements of the new Elgg version.

**Comment by jdalsem** [comment]
This would really be nice to have but takes a lot of effort to create. Best practice is to follow correct upgrade instructions (these instructions will be improved in #10141) and to backup and test.
